### PR TITLE
feat: add workbench page

### LINKED ISSUE #31: Add remote desktop tabs page
Now the assets page (https://door.casvisor.com/assets) allows to open a new browser tab to establish connection to the machine.

The web UI can be further improved by adding a tabs page like below picture, left panel is the list of machines. For right side, tabs of active connected machines will show on the top, the active remote desktop shows on the bottom.

It's also very similar to the experiences of VMware Workstation.

image

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
Fix: https://github.com/casbin/casvisor/issues/31

https://github.com/casbin/casvisor/assets/71440988/9b5f9035-6449-4190-a35d-cd7bb500a418

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -233,6 +192,8 @@ class App extends Component {
     res.push(Setting.getItem(<Link to="/assets">{i18next.t("general:Assets")}</Link>, "/assets"));
     res.push(Setting.getItem(<Link to="/sessions">{i18next.t("general:Sessions")}</Link>, "/sessions"));
     res.push(Setting.getItem(<Link to="/records">{i18next.t("general:Records")}</Link>, "/records"));
+    res.push(Setting.getItem(<Link to="/my-assets">{i18next.t("general:My assets")}</Link>, "/my-assets"));
```
Comment (hsluoyz) [inline web/src/App.js]: Is "/my-assets" still needed?

File: web/src/App.js [hunk 2]
```diff
@@ -54,37 +54,9 @@ class App extends Component {
...
 
-  componentDidUpdate() {
-    // eslint-disable-next-line no-restricted-globals
-    const uri = location.pathname;
-    if (this.state.uri !== uri) {
-      this.updateMenuKey();
-    }
-  }
-
-  updateMenuKey() {
```
Comment (hsluoyz) [inline web/src/App.js]: Why delete `updateMenuKey`? You seem to omit the work done by `.includes`

File: web/src/component/access/AssetTree.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+  const transformData = (data) => {
+    const linuxAssets = [];
+    const windowsAssets = [];
+
+    data.forEach((asset) => {
+      const {protocol, port} = asset;
+
+      if (protocol === "ssh" && port === 22) {
+        linuxAssets.push({
```
Comment (hsluoyz) [inline web/src/component/access/AssetTree.js]: Auto-grouping by Windows and Linux is not very valuable. Using icon to differentiate OS is better. Grouping and orders should be customized by user

File: web/src/App.js [hunk 4]
```diff
@@ -268,71 +229,111 @@ class App extends Component {
...
+    const showMenu = () => {
+      this.setState({
+        menuVisible: true,
+      });
+    };
+
     const onClick = ({key}) => {
-      // eslint-disable-next-line react/prop-types
-      this.props.history.push(key);
+      if (Setting.isMobile()) {
```
Comment (hsluoyz) [inline web/src/App.js]: The diff is not easy to follow here. What does this part of code do?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/App.js (hunk 4 above)
Comment (leo220yuyaodog) [reply web/src/App.js]: `this.props.history.push(key);` casuse a bug that when you click the link that open in a new tab, the current url will also be update. When you click workbench, the current is jump to /workbench in the same time.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**